src/error.jl

```diff
@@ -0,0 +1,29 @@
+"""
+    operation_name(err::Union{UnsupportedError, CannotError})
+
+Return the name of the operation throwing the error in a gerund (i.e. -ing form).
```

Comment:
It's a bit awkward to break up the error message like this. Maybe let each error implement `showerror` instead?

Reply:
This is to enforce consistency in the error message. It also ensure the consistency between the abstract type chosen and the error message. It helps when choosing which supertype should be chosen. For instance, for modify I was hesitant between `UnsupportedError` and `CannotError` but the error should be the one of `UnsupportedError` so that's what I have chosen.